gatorgrouper/utils/group_creation.py

```diff
@@ -0,0 +1,164 @@
+"""Contains all of the gorup creation algorithms"""
```

Comment:
I would suggest adding a single interface method to this module that takes two string constants from `constants.py` (one for grouping method and the other for whether to use the given number for group size or group number) and then calls the correct grouping function based on those arguments. This will make all API and testing calls the same function, enabling better, easier to read code.

Reply:
I agree that there should be a single interface for all the grouping functions. It may be best to use keyword arguments rather than positional arguments so that parameters can be optional and default values can be provided. 

Maybe the function could determine which grouping algorithm should be used (if one is not explicitly specified) based on which keyword arguments are provided? For example, if a "compatibility" parameter is passed in, it would decide to use the graph partitioning algorithm since the others do not take compatibility into account.

Any thoughts on this idea @ALEXANDERB82 @Michionlion ?

Reply:
Sounds good, some requirements should be defined and then this can be designed!